### PR TITLE
Add Component Check to Aggregate YML

### PR DESCRIPTION
**Pull request by alzimmermsft**
This PR adds component check to the aggregate reporting YML.

### CODE REVIEW

**Review by weshaggard — APPROVED** [review]
The changes looks good. You will have to wait until after the PR is merged to verify that the CG step actually works as it is conditioned out for PRs.

File: eng/jacoco-test-coverage/pom.xml
Comment (weshaggard) [inline eng/jacoco-test-coverage/pom.xml:534]: Why is this needed now?

Comment (alzimmermsft) [reply eng/jacoco-test-coverage/pom.xml:534]: This causes a Maven plugin to fail a build because that library isn't using a standard Track 2 groupId, in the parent POM we whitelist and `com.azure:*`, `com.azure.resourcemanager:*`, and `com.azure.spring:*` dependencies as we know those are track 2 and we allow any track 2 library to have a track 2 dependency. This artifact is using `com.microsoft` so it was causing build to fail.

cc: @srnagar

**Review by weshaggard — APPROVED** [review]

File: eng/jacoco-test-coverage/pom.xml
Comment (srnagar) [reply eng/jacoco-test-coverage/pom.xml:534]: We are still debating the namespace for this library. So, I think we can skip this library from coverage checks until the namespace is decided.

**Pull request merged**
